packages/stepper/src/Stepper.tsx

```diff
@@ -0,0 +1,521 @@
+    & > *,
+    &:before,
+    &:after {
+      background: ${uiColors.gray.light3};
+      border-color: ${uiColors.gray.light2};
+    }
+  }
+`;
+
+const hiddenLayerStyle = css`
+  opacity: 0;
+  visibility: hidden;
+  pointer-events: none;
+`;
+
+const slideInAnimationStyle = css`
+  left: 0;
+  transition: left 700ms;
+`;
+
+const slideOutAnimationStyle = css`
+  left: -30px;
+  transition: left 700ms;
+`;
+
+const containerStyle = css`
+  position: relative;
+  height: 46px;
+  width: 100%;
+`;
+
+const stepLabelStyle = css`
+  display: flex;
+  position: absolute;
+
+  align-items: center;
+  justify-content: center;
+  top: calc(100% - 12px);
+
+  height: 24px;
+  min-width: 24px;
+  max-width: max-content;
+
+  font-size: 12px;
+  line-height: 14px;
+
+  border-radius: 12px;
+  border: 1px solid;
+`;
+
+const completedStepLabelStyle = css`
+  background-color: white;
+  border-color: ${uiColors.green.base};
+  color: ${uiColors.green.base};
+`;
+
+const currentStepLabelStyle = css`
+  background-color: ${uiColors.gray.dark2};
+  border-color: #464c4f;
+  color: white;
+`;
+
+const upcomingStepLabelStyle = css`
+  background-color: white;
+  border-color: ${uiColors.gray.light1};
+  color: ${uiColors.gray.base};
+`;
+
+const stepLabelTextStyle = css('padding: 0 8px;');
+
+function InternalStep({
+  children,
+  state,
+  stepLabel,
+  className,
+  ...containerProps
+}: InternalStepProps) {
+  const isCurrent = state === 'current';
+  const isUpcoming = state === 'upcoming';
+  const isCompleted = state === 'completed';
+  return (
+    <div
+      className={cx(
+        stepStyle,
+        { [upcomingStepStyle]: isUpcoming },
+        { [currentStepStyle]: isCurrent },
+        className,
+      )}
+      {...containerProps}
+    >
+      {stepLabel && (
+        <div
+          className={cx(
+            stepLabelStyle,
+            state === 'current'
+              ? currentStepLabelStyle
+              : isCompleted
+              ? completedStepLabelStyle
+              : upcomingStepLabelStyle,
+          )}
+        >
+          {isCompleted ? (
+            <CheckmarkIcon />
+          ) : (
+            <div className={stepLabelTextStyle}>{stepLabel}</div>
+          )}
+        </div>
+      )}
+      {children}
+      <div className={stepBoxStyle}>
+        <span className={stepBoxTopStyle}>
+          <div />
+        </span>
+        <span className={stepBoxBottomStyle}>
+          <div />
+        </span>
+      </div>
+    </div>
+  );
+}
+
+InternalStep.displayName = 'Step';
+
+type StepContents = string | number | Array<StepContents>;
+
+function stepContentsAriaLabel(contents: StepContents): string {
+  return contents instanceof Array
+    ? contents.map(content => stepContentsAriaLabel(content)).join('')
+    : contents.toString();
+}
+
+type StepElement = React.ReactElement<{ children: StepContents }>;
+export type StepElements = StepElement | Array<StepElement>;
+
+interface StepperProps {
```

Comment:
super nit, can we put this right above the Stepper component so they're closer together which makes it easier for me to read?